src/dom.js

```diff
@@ -1,37 +1,40 @@
 import { SVG } from './reconciler'
 
-function updateProperty (dom, name, value, newValue) {
-  if (name === 'style') {
-    for (let k in value) if (!newValue[k]) dom[name][k] = ''
-    for (let k in newValue) dom[name][k] = newValue[k]
-  } else if (name[0] === 'o' && name[1] === 'n') {
-    name = name.slice(2).toLowerCase()
-    if (value) dom.removeEventListener(name, value)
-    dom.addEventListener(name, newValue)
-  } else if (name in dom && !(dom instanceof SVGElement)) {
-    dom[name] = newValue == null ? '' : newValue
-  } else if (newValue == null || newValue === false) {
-    dom.removeAttribute(name)
-  } else {
-    dom.setAttribute(name, newValue)
-  }
-}
+export function updateElement (dom, oldProps, newProps) {
+  const names = Object.keys({...oldProps, ...newProps})
+
+  for (let name of names) {
+    if (name === "children") {
+      continue
+    }
 
-export function updateElement (dom, props, newProps) {
-  Object.keys(newProps)
-    .filter(isNew(props, newProps))
-    .forEach(key => updateProperty(dom, key, props[key], newProps[key]))
+    const oldValue = oldProps[name]
+    const newValue = newProps[name]
+
+    if (name === 'style') {
+      for (let k in oldValue) if (!newValue[k]) dom[name][k] = ''
+      for (let k in newValue) dom[name][k] = newValue[k]
+    } else if (name[0] === 'o' && name[1] === 'n') {
+      name = name.slice(2).toLowerCase()
+      if (oldValue) dom.removeEventListener(name, oldValue)
+      dom.addEventListener(name, newValue)
+    } else if (name in dom && !(dom instanceof SVGElement)) {
+      dom[name] = newValue == null ? '' : newValue
+    } else if (newValue == null || newValue === false) {
+      dom.removeAttribute(name)
+    } else {
+      dom.setAttribute(name, newValue)
+    }
+  }
 }
 
 export function createElement (fiber) {
   const dom =
     fiber.type === 'text'
-      ? document.createTextNode('')
+      ? document.createTextNode(fiber.value)
```

Comment:
Why use fiber.value here, the text nodeValue was in props and update then, so here should be ''?

Reply:
Oh, oops. Good catch! Need to roll back this line change.